File: lib/model/clisitef_data.dart
library clisitef;

import 'package:clisitef/model/data_events.dart';

class CliSiTefData {
  CliSiTefData(
      {required this.event,
        required this.currentStage,
        required this.buffer,
        required this.waiting,
        this.fieldId = 0,
        this.maxLength = 0,
        this.minLength = 0});

  DataEvents event;

  int currentStage;

  String buffer;

  bool waiting = false;

  int fieldId;

  int maxLength;

  int minLength;

  isClientInvoice() {
    return event == DataEvents.data && fieldId == 121;
  }

  isCompanyInvoice() {
    return event == DataEvents.data && fieldId == 122;
  }
}

File: android/src/main/kotlin/com/loopmarket/clisitef/CliSiTefData.kt
package com.loopmarket.clisitef

class CliSiTefData(
    val event: DataEvents,
    val currentStage: Int,
    val buffer: String = "",
    val shouldContinue: Boolean = true,
    val fieldId: Int = 0,
    val maxLength: Int = 0,
    val minLength: Int = 0
) {
    fun toDataSink(): Map<String, Any> {
        return mapOf(
            "event" to event.named,
            "currentStage" to currentStage,
            "buffer" to buffer,
            "shouldContinue" to shouldContinue,
            "fieldId" to fieldId,
            "maxLength" to maxLength,
            "minLength" to minLength
        )
    }
}
